esx_service/utils/auth.py

```diff
@@ -326,6 +326,25 @@ def add_volume_to_volumes_table(tenant_uuid, datastore, vol_name, vol_size_in_MB
 
     return None
 
+def remove_volume_from_volumes_table(tenant_uuid, datastore, vol_name):
+    """ Remove volume from volumes table. """
+    _auth_mgr = get_auth_mgr()
+
+    logging.debug("remove volumes from volumes table(%s %s %s)", tenant_uuid, datastore,
+                  vol_name)
+    try:
+        _auth_mgr.conn.execute(
+                    "DELETE FROM volumes WHERE tenant_id = ? AND datastore = ? AND volume_name = ?", 
+                    [tenant_uuid, datastore, vol_name]
```

Comment:
I checked the schema of this table, I'm a little bit surprised that we are using a combination key of tenant_uuid+datastore+volume_name. Performance is not the main concern here, but I'm concerned we will run into trouble if volumes need to be referenced from other (new introduced) tables.

BTW, our auth proposal document needs to be updated to add this "volumes" table. I can take care of that.